src/main/java/org/mule/tools/maven/plugin/mule/DeployMojo.java

```diff
@@ -166,6 +168,25 @@ public void doExecute() throws MojoExecutionException, MojoFailureException
         }
     }
 
+    private void initializeEnvironment() throws MojoExecutionException
+    {
+        if (server != null)
+        {
+            Server serverObject = this.settings.getServer(server);
+            if (serverObject == null)
+            {
+                getLog().error("Server [" + server + "] not found in settings file.");
+                throw new MojoExecutionException("Server [" + server + "] not found in settings file.");
+            }
+            if (StringUtils.isNotEmpty(username) || StringUtils.isNotEmpty(password))
+            {
+                getLog().warn("Both server and credentials are configured. Using settings.xml server configuration.");
```

Comment:
This should work the other way around, POM settings should take precedence over settings.xml